particles/Native/Kotlin/src/wasm/arcs/WasmParticle.js

```diff
@@ -0,0 +1,145 @@
+defineParticle(({DomParticle, html, log, resolver}) => {
+
+    /**
+     * This looks a bit odd, because it has to fit into the auto-generated setup that Kotlin-native
+     * compiler produces. We can fine-tune this later. We first import the bootstrap
+     * (main.wasm.js) which sets up a bunch of imports for WASM runtime. We add a couple of our
+     * own, including an egregiously bad way of copying strings into WASM because the
+     * current runtine doesn't expose a malloc() callable from JS.
+     *
+     * Also, we never invoke instance.exports.* functions directly, but instead invoke lambda
+     * callbacks exported during startup. Why? Because Kotlin-native's bridging uses its own
+     * pseudo-stack called 'Arenas' to pass arguments by integer reference to WASM. It cleans up
+     * these areas after a lambda callback is finished, reducing the chance of a memory leak.
+     */
+    return class extends DomParticle {
+        constructor() {
+            super()
+            importScripts(resolver(`WasmParticle/main.wasm.js`));
+            this.html = html;
+            this.log = log;
+
+            /**
+             * These are hand written, but many of them eventually can be auto-generated
```

Comment:
What blocks that work? Worth stating in the comment.

Reply:
Blocked on Jetbrains productizing the utility. But I think most of these will go away, since I identified only about 7 functions we really need.